src/modules/shared/helpers/render-og-tags.tsx

```diff
@@ -1,25 +1,28 @@
+import getConfig from 'next/config';
 import Head from 'next/head';
 import { ReactNode } from 'react';
 
 import { createPageTitle } from '@shared/utils';
 
+const { publicRuntimeConfig } = getConfig();
+
 export function renderOgTags(
 	title: string | undefined,
 	description: string,
-	clientUrl: string
+	url: string
 ): ReactNode {
 	const resolvedTitle = createPageTitle(title);
 	return (
 		<Head>
 			<title>{resolvedTitle}</title>
 			<meta name="description" content={description} />
 			<meta property="og:type" content="website" />
-			<meta property="og:url" content={window.location.href} />
+			<meta property="og:url" content={url} />
 			<meta property="og:title" content={resolvedTitle} />
 			<meta property="og:description" content={description} />
-			<meta property="og:image" content={`${clientUrl}/images/og.jpg`} />
+			<meta property="og:image" content={`${publicRuntimeConfig.CLIENT_URL}/images/og.jpg`} />
 			<meta property="twitter:card" content="summary_large_image" />
-			<meta property="twitter:domain" content={window.location.origin} />
+			<meta property="twitter:domain" content={publicRuntimeConfig.CLIENT_URL} />
 			<meta property="twitter:title" content={resolvedTitle} />
 			<meta property="twitter:description" content={description} />
```

Comment:
Basically all of these fields will have to be changed per-page; to avoid parameter bloat I suggest switching to something like the below and providing fallback values.

```
function renderOgTags(config?: {
  // shared
  description?: string;
  
  // og-specific
  og?: {
    type?: string;
    ....
  },
  
  // twitter-specific
  twitter?: { ... }
}): ReactNode { ... }
```

Reply:
> Basically all of these fields will have to be changed per-page

Could be, well implemented it when they ask for it.